ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiUserManagementService.scala

```diff
@@ -100,6 +100,7 @@ private[apiserver] final class ApiUserManagementService(
       } { case (user, pRights) =>
         for {
           _ <- identityProviderExistsOrError(user.identityProviderId)
+          _ <- verifyUserBelongsToIDPOrError(user.identityProviderId, user.id)
```

Comment:
How does that work? We are checking in the db if the user belongs to the idp before we actually add it to the db?

Reply:
So this will actually check that user with this id belongs to the same IDP as the requesting user. If it does not - i.e. there is a user with another "identity_provider_id" - we throw permission denied rather that user exists error. This is the case we discussed before, if user does not exist - this check will pass.

Reply:
Can we then rename this method to better reflect what it does i.e. follow the example of Party Service: verifyUserIsNonExistentOrInIdp

Reply:
sure, Pawel suggested the name for party management service and I forgot to update this one.